### PR TITLE
skip setState if component is unmounted (closes #62)

### LINKED ISSUE #62: Warning shown when component is unmounted while refetch request is pending
If a user clicks on a button that unmounts the current component while the component's refetch request is pending, the following warning will be shown in the console after the refetch request is complete:

`Warning: setState(...): Can only update a mounted or mounting component. This usually means you called setState() on an unmounted component. This is a no-op. Please check the code for the Refetch.connect(PostList) component.`

Are we supposed to cancel the refetch request within `componentWillUnmount()` ? How can we do that?

### PR DESCRIPTION
**Pull request by nfcampos**

### CODE REVIEW

File: src/components/connect.js [hunk 1]
```diff
@@ -253,6 +254,9 @@ export default function connect(mapPropsToRequestsToProps, options = {}) {
       }
 
       setAtomicState(prop, startedAt, mapping, datum, refreshTimeout, callback) {
+        if (this._unmounted)
```
Comment (ryanbrainard) [inline src/components/connect.js]: curly braces please :)

Comment (nfcampos) [inline src/components/connect.js]: added :)

File: test/components/connect.spec.js [hunk 2]
```diff
@@ -949,6 +949,56 @@ describe('React', () => {
...
+          super()
+          this.state = { render: true }
+        }
+
+        setRender(render) {
+          this.setState({ render })
+        }
+
+        render() {
+          if (this.state.render)
```
Comment (ryanbrainard) [inline test/components/connect.spec.js]: more curlies

Comment (nfcampos) [inline test/components/connect.spec.js]: oops, bad habits :)

**Comment by ryanbrainard** [comment]
@nfcampos I think you're updating the docs on the wrong branch...

**Comment by nfcampos** [comment]
yeah after a while I noticed that ;) all fixed now

**Comment by ryanbrainard** [comment]
I already merged the `unmounted` changes into master, but GitHub isn't closing this automatically because I rebased. Closing manually.

@nfcampos Thanks for fixing this!

**Comment by ryanbrainard** [comment]
@nfcampos Just tested this on an internal app that was running into #62 when components were being rapidly mounted/unmounted and it works great now. Thanks! cc: @jsullivan

**Comment by nfcampos** [comment]
@ryanbrainard glad to hear it works!
unrelated: should we add this eslint rule to warn about the curly braces? http://eslint.org/docs/rules/curly

**Comment by ryanbrainard** [comment]
Yes, we should.

**Comment by ryanbrainard** [comment]
Released in https://github.com/heroku/react-refetch/releases/tag/v0.8.0-beta.0

**Pull request closed without merging**